core/src/main/java/org/apache/calcite/sql/SqlUpdate.java

```diff
@@ -43,7 +43,7 @@ public class SqlUpdate extends SqlCall {
   @Nullable SqlSelect sourceSelect;
   @Nullable SqlIdentifier alias;
 
-  @Nullable SqlNode fromClause;
+  @Nullable SqlNode from;
```

Comment:
I think we might want to undo these changes to Calcite entirely. It looks like `SqlUpdate` allows us to extend it so it might be better if we create this class in our own repository and use it there. We can make an `SqlUpdate` that extends this one and adds the additional parameter.

We can override the `createSourceSelectForUpdate` in our own internal implementation of the validator and perform the logic for the creation of the joins in there. This would allow us to implement the behavior without modifying the fork and would allow us to integrate with future versions of Calcite more easily.